ambari-server/src/main/java/org/apache/ambari/server/state/quicklinksprofile/DefaultQuickLinkVisibilityController.java

```diff
@@ -57,6 +68,39 @@ public DefaultQuickLinkVisibilityController(QuickLinksProfile profile) throws Qu
     if (filterCount == 0) {
       throw new QuickLinksProfileEvaluationException("At least one filter must be defined.");
     }
+
+    // compute url overrides
+    profile.getFilters().stream()
+      .filter(f -> f instanceof LinkNameFilter && ((LinkNameFilter)f).getLinkUrl() != null)
+      .findAny()
+      .ifPresent(f -> LOG.warn("Link url overrides only work on service and component levels. Global filter: {}", f));
+    for (Service service : profile.getServices()) {
+      urlOverrides.putAll(getUrlOverrides(service.getName(), service.getFilters()));
+
+      for (Component component : service.getComponents()) {
+        Map<Pair<String, String>, String> componentUrlOverrides = getUrlOverrides(service.getName(), component.getFilters());
+        Sets.SetView<Pair<String, String>> duplicateOverrides = Sets.difference(urlOverrides.keySet(), componentUrlOverrides.keySet());
+        if (!duplicateOverrides.isEmpty()) {
+          LOG.warn("Duplicate url overrides in quick links profile: {}", duplicateOverrides);
+        }
+        urlOverrides.putAll(componentUrlOverrides);
+      }
+    }
+  }
+
+  private Map<Pair<String, String>, String> getUrlOverrides(String serviceName, Collection<Filter> filters) {
+    return filters.stream()
+      .filter( f -> f instanceof LinkNameFilter && null != ((LinkNameFilter)f).getLinkUrl() )
+      .map( f -> {
+        LinkNameFilter lnf = (LinkNameFilter)f;
```

Comment:
Can you please extract the following logic (to `LinkNameFilter`)?  It would allow simplifying the streams in `getUrlOverrides` and the controller constructor.

```
  public static Stream<LinkNameFilter> onlyLinkNameFilters(Stream<? extends Filter> streamOfFilters) {
    return streamOfFilters
      .filter(f -> f instanceof LinkNameFilter)
      .map(f -> (LinkNameFilter) f);
  }
```